src/libraries/Microsoft.PowerFx.Connectors/Execution/FormulaValueSerializer.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.OpenApi.Models;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Connectors.Execution
+{
+    internal abstract class FormulaValueSerializer
+    {
+        internal abstract string GetResult();
+
+        internal abstract void StartSerialization(string referenceId);
+
+        internal abstract void EndSerialization();
+
+        protected abstract void StartObject(string name = null);
+
+        protected abstract void EndObject(string name = null);
+
+        protected abstract void StartArray(string name = null);
+
+        protected abstract void StartArrayElement(string name);
+
+        protected abstract void EndArray();
+
+        protected abstract void WritePropertyName(string name);
+
+        protected abstract void WriteNullValue();
+
+        protected abstract void WriteNumberValue(double numberValue);
+
+        protected abstract void WriteStringValue(string stringValue);
+
+        protected abstract void WriteBooleanValue(bool booleanValue);
+
+        protected abstract void WriteDateTimeValue(DateTime dateTimeValue);
+
+        protected bool _schemaLessBody;
```

Comment:
>protected [](http://example.com/codeflow?start=8&length=9)

nit: readonly. 
(we don't want derived class changing it) 

Although ... why do we need this?  (a comment would be good). 
Does this mean we don't have a OpenApiSchema schema parameter?